cli.js

```diff
@@ -158,7 +158,12 @@ function exit(results) {
 	// correctly flush the output when multiple test files
 	process.stdout.write('');
 
-	process.exit(failed > 0 ? 1 : 0);
+	// Individually kill each child process.
+	Promise.map(results, function (result) {
+		return result.kill();
```

Comment:
`result.kill !== process.kill`. It is a method I attached to the result. It returns a Promise that resolves with `childProcess.on('exit',...`. This is waiting for all the child processes to exit before killing the parent.

`result.kill` also makes sure `process.kill` is not being called on any processes that have already exited. Because: 

> May emit an 'error' event when the signal cannot be delivered. **Sending a signal to a child process that has already exited is not an error but may have unforeseen consequences**: if the PID (the process ID) has been reassigned to another process, the signal will be delivered to that process instead. What happens next is anyone's guess.
>
> **Note that while the function is called kill, the signal delivered to the child process may not actually kill it**. kill really just sends a signal to a process.
>
> ---- https://nodejs.org/api/child_process.html#child_process_child_kill_signal



`process.kill` may return synchronously, but we want to wait for children to actually exit. `exit` events certainly are not processed synchronously. 

Proof: https://github.com/jamestalmage/process-kill-is-sync